k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapFolderPusher.java

```diff
@@ -120,6 +120,44 @@ private void superHandleUntaggedResponse(ImapResponse response) {
         super.handleUntaggedResponse(response);
     }
 
+    /**
+     * Ensure the DONE continuation is only sent when the IDLE command was sent and hasn't completed yet.
+     */
+    private static class IdleStopper {
```

Comment:
unrelated changes moving things about. again, please keep those in separate commits.